src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -463,3 +464,29 @@ void CoreWrap::add_extension(const Napi::CallbackInfo& info) {
         reportError(info.Env(), err.what());
     }
 }
+
+Napi::Value CoreWrap::query_model(const Napi::CallbackInfo& info) {
+    std::vector<std::string> allowed_signatures;
+    try {
+        if (ov::js::validate<ModelWrap, Napi::String>(info, allowed_signatures) || 
+            ov::js::validate<ModelWrap, Napi::String, Napi::Object>(info, allowed_signatures)) {
+            ov::AnyMap properties;
+            auto model = Napi::ObjectWrap<ModelWrap>::Unwrap(info[0].ToObject())->get_model();
+            auto device_name = info[1].ToString();
+            if (info.Length() == 3) {
+                properties = to_anyMap(info.Env(), info[2]);
+            }
+            auto query_result = _core.query_model(model, device_name, properties);
+            Napi::Object result = Napi::Object::New(info.Env());
+            for (const auto& elem: query_result) {
```

Comment:
```suggestion
            for (const auto& elem : query_result) {
```
Beside this, formatting looks good

Reply:
Done